### PR TITLE
Support ICX on windows

### PR DESCRIPTION
**Pull request by Iliamish**
Signed-off-by: Mishin, Ilya <ilya.mishin@intel.com>

### CODE REVIEW

File: src/tbbmalloc/tbbmalloc_internal.h [hunk 1]
```diff
@@ -339,7 +339,14 @@ class BackRefIdx { // composite index to backreference array
...
 class BlockI {
-    intptr_t     blockState[2];
+    #if __clang__
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wunused-private-field"
+        intptr_t     blockState[2];
+        #pragma clang diagnostic pop // "-Wunused-private-field"
+    #else
+        intptr_t     blockState[2];
+    #endif
```
Comment (alexey-katranov) [inline src/tbbmalloc/tbbmalloc_internal.h:349]: The following code has the same implementation for all compilers:
```suggestion
    #if __clang__
        #pragma clang diagnostic push
        #pragma clang diagnostic ignored "-Wunused-private-field"
    #endif
        intptr_t     blockState[2];
    #if __clang__
        #pragma clang diagnostic pop // "-Wunused-private-field"
    #endif
```

File: cmake/compilers/MSVC.cmake [hunk 2]
```diff
@@ -72,4 +72,12 @@ endif()
...
 
-set(TBB_OPENMP_FLAG /openmp)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86|AMD64)")
+        set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -mrtm -mwaitpkg)
+    endif()
+    set(TBB_OPENMP_LINK_FLAG " ")
+    set(TBB_OPENMP_COMPILE_FLAG /openmp)
+else()
+    set(TBB_OPENMP_COMPILE_FLAG /openmp)
```
Comment (alexey-katranov) [inline cmake/compilers/MSVC.cmake:82]: The same option is set in the both branches

File: cmake/compilers/Clang.cmake [hunk 3]
```diff
@@ -21,6 +21,9 @@ elseif (APPLE)
 
     # For correct ucontext.h structures layout
     set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -D_XOPEN_SOURCE)
+elseif(MSVC)
```
Comment (AlexVeprev) [inline cmake/compilers/Clang.cmake:24]: Style:
```suggestion
elseif (MSVC)
```

File: cmake/compilers/Clang.cmake [hunk 4]
```diff
@@ -46,7 +49,7 @@ if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86|AMD64)")
 endif()
 
 if (NOT MINGW)
-    set(TBB_COMMON_LINK_LIBS dl)
+    set(TBB_COMMON_LINK_LIBS ${CMAKE_DL_LIBS})
```
Comment (AlexVeprev) [inline cmake/compilers/Clang.cmake:52]: Since you use `${CMAKE_DL_LIBS}` maybe you don't need `if` for MinGW

File: cmake/compilers/MSVC.cmake [hunk 5]
```diff
@@ -72,4 +72,11 @@ endif()
 set(TBB_IPO_COMPILE_FLAGS $<$<NOT:$<CONFIG:Debug>>:/GL>)
 set(TBB_IPO_LINK_FLAGS $<$<NOT:$<CONFIG:Debug>>:-LTCG> $<$<NOT:$<CONFIG:Debug>>:-INCREMENTAL:NO>)
 
-set(TBB_OPENMP_FLAG /openmp)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86|AMD64)")
```
Comment (AlexVeprev) [inline cmake/compilers/MSVC.cmake:76]: You implement similar logic by different ways, consider:
```cmake
if (CMAKE_CXX_COMPILER_ID MATCHES "(Clang|IntelLLVM)")
```

File: cmake/compilers/MSVC.cmake [hunk 6]
```diff
@@ -72,4 +72,11 @@ endif()
...
+        set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -mrtm -mwaitpkg)
+    endif()
+    set(TBB_OPENMP_LINK_FLAG " ")
```
Comment (AlexVeprev) [inline cmake/compilers/MSVC.cmake:79]: Is it really needed? What if just skip setting this flag?

Hmm, but you'll likely have to put setting `LINK_FLAGS` property under `if (TBB_OPENMP_LINK_FLAG)`

Comment (AlexVeprev) [reply cmake/compilers/MSVC.cmake:79]: Never mind

**Review by alexey-katranov — APPROVED** [review]
I am Ok with the patch. Check with @AlexVeprev that he is also Ok.

**Review by AlexVeprev — APPROVED** [review]

**Review by AlexVeprev — APPROVED** [review]
CMake looks good

**Pull request merged**